src/sources/eventstoredb/types.rs

```diff
@@ -0,0 +1,345 @@
+use crate::event::metric::{MetricData, MetricName, MetricSeries};
+use crate::event::{Metric, MetricKind, MetricValue};
+use crate::internal_events::InternalEvent;
+use hyper::client::HttpConnector;
+use hyper::Body;
+use hyper_openssl::HttpsConnector;
+use metrics::gauge;
+use serde::de::{MapAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::collections::BTreeMap;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Stats {
+    pub proc: Proc,
+    pub sys: Sys,
+}
+
+impl Stats {
+    pub fn metrics(&self, namespace: Option<String>) -> Vec<Metric> {
+        let mut result = Vec::new();
+        let mut tags = BTreeMap::new();
+        let namespace = namespace.unwrap_or_else(|| "eventstoredb".to_string());
+
+        tags.insert("id".to_string(), self.proc.id.to_string());
+
+        let name = MetricName {
+            name: "memory_usage".to_string(),
+            namespace: Some(namespace.clone()),
+        };
+
+        let data = MetricData {
+            timestamp: None,
+            kind: MetricKind::Absolute,
+            value: MetricValue::Gauge {
+                value: self.proc.mem as f64,
+            },
+        };
+
+        let series = MetricSeries {
+            name,
+            tags: Some(tags.clone()),
+        };
+
+        result.push(Metric { series, data });
```

Comment:
`Metric` can also be builded like 
```
Metric::new(
    "memory_usage",
    MetricKind::Absolute,
    MetricValue::Gauge { value: self.proc.mem as f64 },
)
.with_namespace(namespace.clone())
.with_tags(Some(tags.clone()))
.with_timestamp(Some(now))
```

if you want to shorten this.

Reply:
Done